lib/browserid.js

```diff
@@ -68,28 +80,25 @@ module.exports = function verifyAssertion(assertion) {
     if (! body || body.status !== 'okay') {
       return error('non-okay response', body);
     }
-    var email = body.email;
-    var parts = email.split('@');
-    var allowedIssuer = config.get('browserid.issuer');
+    const email = body.email;
+    const parts = email.split('@');
+    const allowedIssuer = config.get('browserid.issuer');
     if (parts.length !== 2 || parts[1] !== allowedIssuer) {
       return error('invalid email', email);
     }
     if (body.issuer !== allowedIssuer) {
       return error('invalid issuer', body.issuer);
     }
-    var uid = parts[0];
-    //sanity check the uid
-    if (uid.length !== 32 && HEX_STRING.test(uid)) {
```

Comment:
I'm pretty sure this old validation code was incorrect, and should have been `|| ! HEX_STRING.test(uid)`.  I've added some tests for the validation here, and moved it into a Joi schema so it's more declarative.